jetstream/statistics.py

```diff
@@ -115,13 +115,17 @@ def apply(
         statistic_result_collection = StatisticResultCollection([])
 
         if metric in df:
-            branch_list = df.branch.to_numpy()
-            if reference_branch not in branch_list or reference_branch is None:
+            branch_list = df.branch.unique()
+            if reference_branch and reference_branch not in branch_list:
                 logging.warn(f"Branch {reference_branch} not in {branch_list} for {self.name()}.")
             else:
-                statistic_result_collection.data += self.transform(
-                    df, metric, reference_branch
-                ).data
+                if reference_branch is None:
+                    ref_branch_list = branch_list
```

Comment:
Half of this work is redundant, right? The comparison (a, b) is the same as -(b, a). @emtwo Do you think we should compute all comparisons both ways to make life easier for the visualization layer?

Reply:
Is there an example of the comparison of a to b vs. b to a? I imagine there might be some computation to "invert" the value, depending on what the statistic is. If this is the case, then the frontend will need to have some logic for that. In which case, for the sake of completeness of the data and keeping the data processing out of the front end, I think the redundancy would be fine here.

Reply:
Like if I'm comparing message A to message B and message A does 5% better on some metric than message B, that means that message B did 4.8% (1-1/1.05) worse than message A. I think there should be a simple mechanical relationship in all cases.

There's maybe a cosmetic problem if we compute both (a, b) and (b, a) insofar as the bootstrap is a stochastic procedure and the results for each dupe may be slightly different, which might upset people.

Reply:
Then would it make sense to compute only one of (a, b) and (b, a) using the bootstrap procedure and the inverse is computed with the mechanical relationship?